### PR TITLE
Add some PR checking scripts

### PR DESCRIPTION
**Pull request by tellison**
Check assertions etc as part of a PR build job.

Intended to be called from Jenkins as part of a PR builder job, the script `jenkins/run-pr-tests.py` picks up environment variables, runs some basic PR checks (empty right now), and reports back to the PR on the status of the checks.  Returns a non-zero code if checks fail so that Jenkins knows the job should be marked as failed.  New checks can be added easily.

### CODE REVIEW

File: jenkins/pr_tests/pr_check_author.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+def pr_check_author(author):
+    """
+    Check that the author is acceptable for this PR.
+    This could check they are not on a banned list, or that they
+    do have an OCA, etc. etc.
+
+    returns a tuple, [boolean, string] = [whether the check passed, a message to display in the PR]
+    """
+    print("Bogus author check")
```
Comment (karianna) [inline jenkins/pr_tests/pr_check_author.py]: Maybe "unauthorised author check" ?

File: jenkins/pr_tests/pr_check_author.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    if author == "untrustedguy":
+        print("Failed author check : %s" % author)
```
Comment (karianna) [inline jenkins/pr_tests/pr_check_author.py]: "untrustedperson"

File: jenkins/run-pr-tests.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    msg : String message
+    ghprb_pull_id : Integer Github pull request identifier
+    """
+
+    print("Posting message to Github PR %s" % ghprb_pull_id)
+
+    posted_message = json.dumps({"body": msg})
+
+    url = "https://api.github.com/repos/adoptopenjdk/openjdk-jdk8u/issues/" + ghprb_pull_id + "/comments"
+
```
Comment (karianna) [inline jenkins/run-pr-tests.py]: environmentalise "openjdk-jdk8u"?

File: jenkins/run-pr-tests.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    ...etc...
+    """
+
+    # Save off the current HEAD (for revert)
+    current_pr_head = run_cmd(['git', 'rev-parse', 'HEAD'], return_output=True).strip()
+    pr_all_checks_pass = True
+    pr_all_checks_results = list()
+
+    print("Checking Foo")
+    # Do a foo check here
```
Comment (karianna) [inline jenkins/run-pr-tests.py]: Is this example code?  I'm not sure of Foo :-)

File: jenkins/run-pr-tests.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+
+def main():
+    ghprb_pull_id = os.environ["ghprbPullId"]
+    ghprb_actual_commit = os.environ["ghprbActualCommit"]
+    ghprb_pull_title = os.environ["ghprbPullTitle"]
+    build_display_name = os.environ["BUILD_DISPLAY_NAME"]
+    build_url = os.environ["BUILD_URL"]
+    commit_url = "https://github.com/adoptopenjdk/openjdk-jdk8u/commit/" + ghprb_actual_commit
+
```
Comment (karianna) [inline jenkins/run-pr-tests.py]: again environmentalise openjdk-jdk8u

File: jenkins/pr_tests/pr_check_author.py (hunk 1 above)
Comment (tellison) [inline jenkins/pr_tests/pr_check_author.py]: Right, this is sample code to test the Github integration options.
Still requires writing the actual checks we are interested in performing on each PR.  We can discuss what they should be doing.

File: jenkins/pr_tests/pr_check_author.py (hunk 2 above)
Comment (tellison) [inline jenkins/pr_tests/pr_check_author.py]: Clearly a hard-coded username is not the answer here ;-)  For demo purposes.

File: jenkins/run-pr-tests.py (hunk 3 above)
Comment (tellison) [inline jenkins/run-pr-tests.py]: Yep, can use `ghprbGhRepository`

File: jenkins/run-pr-tests.py (hunk 4 above)
Comment (tellison) [inline jenkins/run-pr-tests.py]: :-)   yes, this is example code

**Comment by tellison** [comment]
Apologies for not being clear on the status of this PR.  It is not conducting any useful PR checks at the moment (hence the "runs some basic PR checks (empty right now)" comment).

If people think this is a reasonable framework for conducting PR checks then we can build upon it, but if I'm heading up the wrong way then I'll stop now!

File: jenkins/pr_tests/pr_check_author.py (hunk 1 above)
Comment (karianna) [inline jenkins/pr_tests/pr_check_author.py]: Cool, can you add a TODO comment here in that case?

File: jenkins/pr_tests/pr_check_author.py (hunk 2 above)
Comment (karianna) [inline jenkins/pr_tests/pr_check_author.py]: TOOD comment here as well

File: jenkins/run-pr-tests.py (hunk 3 above)
Comment (karianna) [inline jenkins/run-pr-tests.py]: Make that change now?

File: jenkins/run-pr-tests.py (hunk 4 above)
Comment (karianna) [inline jenkins/run-pr-tests.py]: TODO comment here

**Pull request closed without merging**